proposals/NNNN-unicode-scalar-properties.md

```diff
@@ -0,0 +1,370 @@
+  public var isIdeographic: Bool { get }    // UCHAR_IDEOGRAPHIC
+  public var isIDSBinaryOperator: Bool { get }    // UCHAR_IDS_BINARY_OPERATOR
+  public var isIDSTrinaryOperator: Bool { get }    // UCHAR_IDS_TRINARY_OPERATOR
+  public var isJoinControl: Bool { get }    // UCHAR_JOIN_CONTROL
+  public var isLogicalOrderException: Bool { get }    // UCHAR_LOGICAL_ORDER_EXCEPTION
+  public var isLowercase: Bool { get }    // UCHAR_LOWERCASE
+  public var isMath: Bool { get }    // UCHAR_MATH
+  public var isNoncharacterCodePoint: Bool { get }    // UCHAR_NONCHARACTER_CODE_POINT
+  public var isQuotationMark: Bool { get }    // UCHAR_QUOTATION_MARK
+  public var isRadical: Bool { get }    // UCHAR_RADICAL
+  public var isSoftDotted: Bool { get }    // UCHAR_SOFT_DOTTED
+  public var isTerminalPunctuation: Bool { get }    // UCHAR_TERMINAL_PUNCTUATION
+  public var isUnifiedIdeograph: Bool { get }    // UCHAR_UNIFIED_IDEOGRAPH
+  public var isUppercase: Bool { get }    // UCHAR_UPPERCASE
+  public var isWhitespace: Bool { get }    // UCHAR_WHITESPACE
+  public var isXIDContinue: Bool { get }    // UCHAR_XID_CONTINUE
+  public var isXIDStart: Bool { get }    // UCHAR_XID_START
+  public var isCaseSensitive: Bool { get }    // UCHAR_CASE_SENSITIVE
+  public var isSentenceTerminal: Bool { get }    // UCHAR_S_TERM
+  public var isVariationSelector: Bool { get }    // UCHAR_VARIATION_SELECTOR
+  public var isNFDInert: Bool { get }    // UCHAR_NFD_INERT
+  public var isNFKDInert: Bool { get }    // UCHAR_NFKD_INERT
+  public var isNFCInert: Bool { get }    // UCHAR_NFC_INERT
+  public var isNFKCInert: Bool { get }    // UCHAR_NFKC_INERT
+  public var isSegmentStarter: Bool { get }    // UCHAR_SEGMENT_STARTER
+  public var isPatternSyntax: Bool { get }    // UCHAR_PATTERN_SYNTAX
+  public var isPatternWhitespace: Bool { get }    // UCHAR_PATTERN_WHITE_SPACE
+  public var isCased: Bool { get }    // UCHAR_CASED
+  public var isCaseIgnorable: Bool { get }    // UCHAR_CASE_IGNORABLE
+  public var changesWhenLowercased: Bool { get }    // UCHAR_CHANGES_WHEN_LOWERCASED
+  public var changesWhenUppercased: Bool { get }    // UCHAR_CHANGES_WHEN_UPPERCASED
+  public var changesWhenTitlecased: Bool { get }    // UCHAR_CHANGES_WHEN_TITLECASED
+  public var changesWhenCaseFolded: Bool { get }    // UCHAR_CHANGES_WHEN_CASEFOLDED
+  public var changesWhenCaseMapped: Bool { get }    // UCHAR_CHANGES_WHEN_CASEMAPPED
+  public var changesWhenNFKCCaseFolded: Bool { get }    // UCHAR_CHANGES_WHEN_NFKC_CASEFOLDED
+  public var isEmoji: Bool { get }    // UCHAR_EMOJI
+  public var isEmojiPresentation: Bool { get }    // UCHAR_EMOJI_PRESENTATION
+  public var isEmojiModifier: Bool { get }    // UCHAR_EMOJI_MODIFIER
+  public var isEmojiModifierBase: Bool { get }    // UCHAR_EMOJI_MODIFIER_BASE
+
+  public var isDefined: Bool { get }  // u_isdefined
+  public var hasNormalizationBoundaryBefore: Bool { get }  // unorm2_hasBoundaryBefore
```

Comment:
Either explicitly note that this is using the stdlib's preferred normalization form for comparisons, or include all forms, or remove.

Reply:
Good point. I'll remove it for now (and update my implementation PR soon); in the future we may want to provide more thorough wrappers around the normalizer APIs.